tracer/src/Datadog.Trace/AsyncLocalScopeManager.cs

```diff
@@ -55,6 +55,12 @@ public void Close(Scope scope)
 
             // scope.Parent is null for distributed traces, so use scope.Span.Context.Parent
             DistributedTracer.Instance.SetSpanContext(scope.Span.Context.Parent as SpanContext);
+
+            // Propagate the resource name to the profiler for root web spans
+            if (scope.Parent == null && scope.Span.Type == SpanTypes.Web)
```

Comment:
Why shouldn't we propagate other span type resource names?

Reply:
The specification explicitly limits the feature to web spans.

> In order for this feature to work profiling needs to capture the ‘resource’ field of the ‘local root span’ of each request. For now we are focusing only on spans with type ‘WEB’.